tests/conftest.py

```diff
@@ -1,42 +1,139 @@
 # SPDX-License-Identifier: MIT
 
+import os
+import os.path
+import shutil
+import stat
+import sys
+import tarfile
+import tempfile
+
+import filelock
 import pytest
 
+import build
+
+
+if sys.version_info[0] == 2:
+    from urllib2 import urlopen
+else:
+    from urllib.request import urlopen
+
 
-DUMMY_PYPROJECT = '''
-[build-system]
-requires = ['flit_core']
-build-backend = 'flit_core.buildapi'
-'''.strip()
+INTEGRATION_SOURCES = {
+    'dateutil': ('dateutil/dateutil', '2.8.1'),
+    'pip': ('pypa/pip', '20.2.1'),
+    'Solaar': ('pwr-Solaar/Solaar', '1.0.3'),
+}
 
 
-DUMMY_PYPROJECT_NO_BACKEND = '''
-[build-system]
-requires = []
-'''.strip()
+def pytest_addoption(parser):
+    parser.addoption('--run-integration', action='store_true', help='run the integration tests')
+    parser.addoption('--only-integration', action='store_true', help='only run the integration tests')
```

Comment:
These two should be put within an exclusive group, otherwise passing them at the same time will cause confusion.

Reply:
What do you mean exactly? They are already under a custom group in pytest ("custom options").

Reply:
https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.add_mutually_exclusive_group

Reply:
Ah! That clarifies things, thanks!

Reply:
```
  File "/home/anubis/git/python-build/tests/conftest.py", line 31, in pytest_addoption
    group = parser.add_mutually_exclusive_group()
AttributeError: 'Parser' object has no attribute 'add_mutually_exclusive_group'
```

Can I keep this out? The arguments actually work properly when used together.

Reply:
Can you check why is that? Properly is a bit stretch as you can pass them at the same time and things will work.

Reply:
Why it still works when I pass them together or why the pytest parser is not a argparse instance?

They can be passed at the same time because the logic handles this.

```python
for item in items:
if os.path.basename(item.location[0]) == 'test_integration.py':
    if not config.getoption('--run-integration') and not config.getoption('--only-integration'):
        item.add_marker(skip_int)
    elif config.getoption('--only-integration'):
        item.add_marker(skip_other)
```

I would be okay making them mutually exclusive but also having them together. Having them together seems easier now.

Reply:
I get why they can be passed together, but I'd rather not allow that use case than make it ambigous what happens when you pass it. Both skip or run is valid in that case, so I'd rather raise an error than go ahead. So can you check how to make it exclusive flags with pytest?

Reply:
I added a manual check and raised `pytest.UsageError`.